### PR TITLE
Update PyTorch export tutorial

### PR DESCRIPTION
**Pull request by pavithraes**
This PR updates the "Exporting StableHLO from PyTorch" tutorial. Specifically:

- Update dependencies, combine the tensorflow dependency in the first section to avoid warnings (about using tensorflow-cpu) on Kaggle
- Minor updates to narration and links, along with an introduction statement
- Removed dynamic batch size code, addedd notes instead

(Supersedes #2616, same updates but we get a rich diff here)

### CODE REVIEW

File: docs/tutorials/pytorch-export.ipynb [hunk 1]
```diff
@@ -125,22 +137,31 @@
...
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "# TODO: FIX\n",
+        "# Kernel crashes with \"INVALID_ARGUMENT: Non-broadcast dimensions must not be dynamic\"\n",
```
Comment (GleasonK) [inline docs/tutorials/pytorch-export.ipynb:153]: TorchXLA coverage of dynamic shapes isn't great, for dynamic shape / export workflows we actually recommend torch_xla2 which uses PT->JAX->StableHLO, and has much better coverage for dynamic shapes:
https://github.com/pytorch/xla/tree/master/experimental/torch_xla2

Perhaps we just make a note of "For dynamic shape export we recommend using torch_xla2 (link), which converts PyTorch models to JAX prior to exporting to StableHLO and has high opset coverage along with great dynamic shape support."

If we want to add a tutorial on using torch_xla2, it likely wouldn't be too hard, here's a recent example ([discord invite needed to view](https://discord.gg/ZKXq7b3V8A)):
https://discord.com/channels/999073994483433573/1281512734902714379/1298361762839527467

I believe the following should work:

```
# Install torch_xla2
pip install torch-xla2 # Note: may need to follow GH instructions linked above but pypi package looks up to date.

# Export the program from native pytorch using torch.export
from torch.export import Dim
batch = Dim(\"batch\", min=4, max=6)
dynamic_shapes = ({0: batch},)
dynamic_export = export(resnet18, sample_input, dynamic_shapes=dynamic_shapes)

# Convert the exported program to StableHLO using torch_xla2
from torch_xla2 import export
stablehlo = export.exported_program_to_stablehlo(exported).mlir_module()
print(stablehlo)
```

Note the print will include _lots_ of debug info, so maybe use some indexing, or print starting with the word `@main` for 1k characters or something to keep the tutorial readable. I.e. `print(stablehlo[1000:2000])` or something.

**Review by GleasonK — CHANGES_REQUESTED** [review]

File: docs/tutorials/pytorch-export.ipynb (hunk 1 above)
Comment (pavithraes) [reply docs/tutorials/pytorch-export.ipynb:153]: Thanks for the notes, I'll include it in the PR.

Comment (pavithraes) [reply docs/tutorials/pytorch-export.ipynb]: @GleasonK - I just updated this section with a Tip about `torch_xla2` to unblock merging. We can update the page iteratively based on the following.

The `torch_xla2` code was still failing with:

```python-traceback
RuntimeError: ('No lowering found for\n\nWhile executing %_native_batch_norm_legit_functional : 
...
```
I can share the full traceback here or in a separate issue if you're interested.

Alternatively, I can use a different, simpler example (like on Discord) just for the dynamic shapes section. For this, we can have a separate `torch_xla2` section at the end.
---

Aside, I checked with the team internally, and they identified this as a potential `MaxPoolNd` lowering bug. Victor was able to get dynamic shapes export working with some explicit `maxpool` and `avgpool` settings: [Kaggle notebook with working snippet](https://www.kaggle.com/code/pavithraes/export-pytorch-to-stablehlo-with-dynamic-shapes). 

If the dynamic shapes support is not-stable, I think it's best to exclude code samples. That said, I'm happy to include the working example if you'd like to have it. :)

File: docs/tutorials/pytorch-export.ipynb [hunk 2]
```diff
@@ -123,24 +135,11 @@
...
-        "from torch.export import Dim\n",
-        "\n",
-        "# Create a dynamic batch size, for the first dimension of the input\n",
-        "batch = Dim(\"batch\", max=15)\n",
-        "dynamic_shapes = ({0: batch},)\n",
-        "dynamic_export = export(resnet18, sample_input, dynamic_shapes=dynamic_shapes)\n",
-        "dynamic_stablehlo = exported_program_to_stablehlo(dynamic_export)\n",
-        "print(dynamic_stablehlo.get_stablehlo_text('forward')[0:5000],\"\\n...\")\n",
-        "```"
+        "_`torch_xla` doesn't yet support exporting models with dynamic shapes to StableHLO, so we recommend using the experimental library [`torch_xla2`](https://github.com/pytorch/xla/tree/master/experimental/torch_xla2) for this. It has a similar API for exporting, converts PyTorch models to JAX prior to exporting to StableHLO, and has high opset coverage along with great dynamic shape support._"
```
Comment (GleasonK) [inline docs/tutorials/pytorch-export.ipynb:142]: `torch_xla` support for exporting dynamic models is limited, for these cases we recommend using `torch_xla2` (link). This lowering path leverages JAX for lowering to StableHLO, and has high opset coverage with much broader support for exported programs with dynamic shapes.

**Review by GleasonK — APPROVED** [review]

File: docs/tutorials/pytorch-export.ipynb (hunk 2 above)
Comment (pavithraes) [reply docs/tutorials/pytorch-export.ipynb]: Updated, thanks.

File: docs/tutorials/pytorch-export.ipynb (hunk 1 above)
Comment (GleasonK) [reply docs/tutorials/pytorch-export.ipynb]: Let's use the note for now, repro would be great and can update in the near future. This should be stable :) but if it isn't then lets avoid the code example as you said.

Comment (GleasonK) [reply docs/tutorials/pytorch-export.ipynb]: Could you open repro as a bug ticket on the ptxla repo?
https://github.com/pytorch/xla/tree/master

**Pull request merged**